src/main/java/com/axis/jenkins/plugins/eiffel/eiffelbroadcaster/MQConnection.java

```diff
@@ -205,6 +190,29 @@ private void sendMessages() {
         }
     }
 
+    /**
+     * Start or restart the message queue thread as necessary. Requires that
+     * the MQConnection has been initialized with the needed configuration.
+     *
+     * @return true if the message queue thread was started, otherwise false
+     */
+    private boolean startMessageQueueThread() {
+        synchronized (lock) {
+            if (!initialized || (messageQueueThread != null && messageQueueThread.isAlive())) {
```

Comment:
Hmm.. I wonder if the `initialized` variable can see an old thread cached value since `public void initialize()`is set from a non synchronized block. Making the var volatile should solve that issue. Not sure if it is necessary though.

Reply:
Ah, excellent point! I totally forgot about synchronization of `initialized`. I think you're right that we need volatile, although the consequences of caching would only be that the thread gets to start slightly later, which in most cases won't be a problem. But let's fix it anyway. I've pushed an extra commit. Will squash the branch into a single commit when merging.